test/livebook/intellisense_test.exs

```diff
@@ -1053,6 +1053,42 @@ defmodule Livebook.IntellisenseTest do
              ] = Intellisense.get_completion_items("struct.my", binding, env)
     end
 
+    test "completion for struct keys inside struct" do
```

Comment:
Can we try autocompletion of an exception and make sure the `__exception__` field does not show up? In the Elixir code we reject anything starting with `_` if the hint is empty.

Reply:
I added a test for this, but it's succeeding because we ignore exceptions from autocompletion entirely in `has_struct?`: https://github.com/livebook-dev/livebook/blob/6784fa9b46898189461ffd7d9d17c0d1f8b4424a/lib/livebook/intellisense/identifier_matcher.ex#L201-L204

This means that we don't provide autocompletion for exception fields when trying to access them:

<img width="969" alt="Screen Shot 2021-12-11 at 5 18 14 PM" src="https://user-images.githubusercontent.com/34720/145693463-6afb3e3d-3a8e-4008-8941-b1940efa7663.png">

To be honest I don't understand why that check is there

Reply:
I think my last line wasn't clear: I think completing `message` in this screenshot above is more correct and would match IEx, but I don't want to remove that check in `has_struct?` because I don't understand what it's for.

Reply:
@jonatanklosko is there a reason why we exclude exceptions?

Reply:
We match the IEx behaviour here:

![image](https://user-images.githubusercontent.com/17034772/145711189-a2b07a9f-e650-4e16-a601-8b1c53a03297.png)

```elixir
iex(2)> e = %ArgumentError{}
%ArgumentError{message: "argument error"}
iex(3)> e.
__exception__    __struct__       message 
```

The exclusion is for completing struct creation:

![image](https://user-images.githubusercontent.com/17034772/145711202-694cce74-e2c5-41ae-9349-446b1c95775d.png)

---

Note: in order to complete map values we need to be in a new cell, so that we have access to that value.

Reply:
Ah so I used has_sruct in my code which now causes us to not autocomplete any keys inside exception structs

I'll pull apart the two checks so we're correct here